internal/dataplane/parser/translate_tcproute.go

```diff
@@ -137,54 +140,64 @@ func generateKongRoutesFromTCPRouteRule(tcproute *gatewayv1alpha2.TCPRoute, rule
 		},
 	}
 
-	routes = append(routes, r)
-
-	return routes, nil
+	return append(routes, r), nil
 }
 
 // generateKongServiceFromTCPRouteBackendRef converts a provided backendRef for an TCPRoute
 // into a kong.Service so that routes for that object can be attached to the Service.
 // TODO add a generic backendRef handler for all GW routes. HTTPRoute needs a wrapper because it uses a special wrapped
 // type with filters. Deferred til after https://github.com/Kong/kubernetes-ingress-controller/issues/2166 though
 // we probably shouldn't see much change to the service (just the upstream it references in Host)
```

Comment:
I'd originally wanted to incorporate this change into the update for multiple BackendRefs, but since the rest of the PR is unaffected by this and you're going on PTO soon, not requesting an actual change for this beyond creating a proper issue and removing the outdated comment. I plan to handle it alongside updating UDPRoute for #2405, or TLSRoute if the UDPRoute change is largely done but not yet PRed.

```suggestion
// generateKongServiceFromTCPRouteBackendRef converts a provided backendRef for an TCPRoute
// into a kong.Service so that routes for that object can be attached to the Service.
// TODO https://github.com/Kong/kubernetes-ingress-controller/issues/2421 add a generic backendRef handler for all GW routes
```